### DISCUSSION TITLE
TypeScript (.ts) and ESM (.mjs) config file support

### DISCUSSION THREAD

**Discussion by halvardssm** [remix-run/remix#2795] [Proposals]
Hi! I am adopting Remix into my new app, but I was wondering if it would be possible to support the remix config file written in TypeScript or ESM?

**Comment by sergiodxa** [comment]
The remix.config file only supports JS and CommonJS right now, it’s directly required by the CLI I believe.

There was a few discussions before about adding TS support, but team decided to don’t go that way (I don’t remember the reason but you could search in the closed issues and PRs),.

**Comment by kiliman** [comment]
You can always create a pre-build step to compile the .ts/.mjs file to .js file.

It's such a small file and once you set the config, you rarely touch it again. So I'm not sure it's worth the effort.
